### PR TITLE
Async Docs

### PR DESCRIPTION
**Pull request by pgjones**
See commits

### CODE REVIEW

File: docs/async-await.rst [hunk 1]
```diff
@@ -39,6 +39,21 @@ most use cases, but Flask's async support enables writing and using
...
 
 
+Background tasks
+----------------
+
+Async functions will run in an event loop until they complete, at
+which stage the event loop will stop. This means any additional
+spawned tasks that haven't completed when the async function completes
+will be cancelled. Therefore you cannot spawn background tasks, for
+example via ``asyncio.ensure_future``.
```
Comment (graingert) [inline docs/async-await.rst:49]:
```suggestion
example via ``asyncio.create_task``
```

File: docs/async-await.rst [hunk 2]
```diff
@@ -39,6 +39,21 @@ most use cases, but Flask's async support enables writing and using
 code that wasn't possible natively before.
 
 
+Background tasks
+----------------
+
+Async functions will run in an event loop until they complete, at
+which stage the event loop will stop. This means any additional
+spawned tasks that haven't completed when the async function completes
+will be cancelled. Therefore you cannot spawn background tasks, for
```
Comment (graingert) [inline docs/async-await.rst:48]: Will they be cancelled or just stopped?

Comment (graingert) [reply docs/async-await.rst:48]: ah yes they are cancelled https://github.com/django/asgiref/blob/main/asgiref/sync.py#L240

File: docs/async-await.rst (hunk 1 above)
Comment (pgjones) [reply docs/async-await.rst:49]: Ta

**Comment by jab** [comment]
Thanks for the great docs!

I tried to preview them rendered as html, but the readthedocs.org CI failed, looks like due to the following?
```
WARNING: autodoc: failed to import class 'Flask' from module 'flask'; the following exception was raised:
cannot import name 'ContextVar' from 'werkzeug.local' (/home/docs/checkouts/readthedocs.org/user_builds/flask/envs/3967/lib/python3.7/site-packages/werkzeug/local.py)
```

**Comment by davidism** [comment]
It should be fixed because it depends on Werkzeug rc now, maybe I need to clear the cache on rtd. It should build locally though

**Comment by jab** [comment]
Thanks @davidism, confirmed the docs build just fine for me locally. And looks like other builds are working just fine on rtd too (see e.g. https://flask.palletsprojects.com/en/master/async-await/). I can submit [a trivial docs PR](https://github.com/pallets/flask/compare/master...jab:docs?expand=1) if it'd help to confirm that the rtd PR docs builds are working again.

**Pull request merged**